gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/license/domain_service/LicenseDomainService.java

```diff
@@ -0,0 +1,79 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.license.domain_service;
+
+import static java.util.Map.entry;
+
+import io.gravitee.apim.core.debug.model.ApiDebugStatus;
+import io.gravitee.apim.core.environment.crud_service.EnvironmentCrudService;
+import io.gravitee.apim.core.environment.model.Environment;
+import io.gravitee.apim.core.event.crud_service.EventCrudService;
+import io.gravitee.apim.core.event.model.Event;
+import io.gravitee.apim.core.license.crud_service.LicenseCrudService;
+import io.gravitee.apim.core.license.model.License;
+import io.gravitee.rest.api.model.EventType;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class LicenseDomainService {
+
+    private final LicenseCrudService licenseCrudService;
+    private final EventCrudService eventService;
+    private final EnvironmentCrudService environmentCrudService;
+
+    public LicenseDomainService(
+        LicenseCrudService licenseCrudService,
+        EventCrudService eventService,
+        EnvironmentCrudService environmentCrudService
+    ) {
+        this.licenseCrudService = licenseCrudService;
+        this.eventService = eventService;
+        this.environmentCrudService = environmentCrudService;
+    }
+
+    public Optional<License> getLicenseByOrganizationId(String organizationId) {
+        return this.licenseCrudService.getOrganizationLicense(organizationId);
+    }
+
+    public License createOrUpdateOrganizationLicense(String organizationId, String license) {
+        var organizationLicense =
+            this.licenseCrudService.getOrganizationLicense(organizationId)
+                .map(l -> this.licenseCrudService.updateOrganizationLicense(organizationId, license))
+                .orElse(this.licenseCrudService.createOrganizationLicense(organizationId, license));
+
+        this.pushOrganizationLicenseUpdateEvent(organizationId, license);
+        return organizationLicense;
+    }
+
+    private void pushOrganizationLicenseUpdateEvent(String organizationId, String license) {
+        var environmentIds = environmentCrudService
+            .findByOrganizationId(organizationId)
+            .stream()
+            .map(Environment::getId)
+            .collect(Collectors.toSet());
+
+        eventService.createEvent(
+            organizationId,
+            null, // FIXME. Environment ID is used to set in execution context. Is that OK to pass null ?
```

Comment:
I don't know 😁

Reply:
yes good for me, it only used to find a user later